### ISSUE TITLE
[支持请求]当节点是“ hysteria2”无法使用 <title>

### ISSUE THREAD

**Issue by gavin8282** [daeuniverse/dae#599]
### Checks

- [X] I have searched the existing issues
- [X] I have read the documentation
- [X] Is it your first time sumbitting an issue

### Support Request

我的使用场景是“回国”，服务端为“hysteria2”，入股dae配置hysteria2服务节点是无法使用的。只能使用hysteria2客户端，然后使用socks5模式

### Current Behavior

我在国内搭建了hysteria2服务
服务端配置如下：
listen: :12443

tls:
  cert:/SSL/hy.123.com/dns-certs.pem
  key: /SSL/hy.123.com/dns-key.pem

auth:
  type: password
  password: 1234567890

masquerade:
  type: proxy
  proxy:
    url: https://www.bing.com/
    rewriteHost: true

bandwidth:
    up: 100 mbps
    down: 100 mbps

国外的大鹅配置如下：
global {
    lan_interface: ens3
    wan_interface: auto
    log_level: debug
    auto_config_kernel_parameter: true
    dial_mode: domain
    allow_insecure: false
    so_mark_from_dae: 1234
    tls_implementation: utls
    utls_imitate: chrome_auto
}
subscription {
}
node {
    # 节点URI
    CN: 'hysteria2://1234567890@hy.123.com:12443'  ##无法连接hy服务器，
}
group {
    cn_proxy {
        policy: fixed(0)
    }
}
dns {
    upstream {
        alidns: 'tcp://223.5.5.5:53'
        googledns: 'udp://8.8.8.8:53'
    }
  routing {
    request {
      qname(geosite:cn) -> alidns
      fallback: googledns
    }
    response {
        upstream(googledns) -> accept
        fallback: accept
    }
  }
}
routing {
    #pname(hysteria) -> must_direct
    pname(NetworkManager) -> direct
    pname(systemd-networkd) -> direct
    dip(224.0.0.0/3, 'ff00::/8') -> direct
    dip(geoip:private) -> direct
    dip(8.8.8.8, 8.8.4.4) -> direct
    dip(223.5.5.5, 119.29.29.29, 114.114.114.114) -> cn_proxy
    #ip(geoip:cn) -> cn_proxy
    domain(geosite:cn) -> cn_proxy
    domain(geosite:category-ads) -> block
        fallback: direct
}

### Expected Behavior

不知道我的方法是否正确，请大神们指正

### Steps to Reproduce

场景1：正常
启动hy2客户端，dae使用hy2的socks5
场景2：无法连接服务器
dae直接使用hysteria2://1234567890@hy.123.com:12443，无法连接到服务器
临时发现的解决方案：
1、先启动一次hy2客户端，连接服务器后结束该服务；
2、dae直接使用hysteria2://1234567890@hy.123.com:12443，成功连接。

### Environment

- **Dae version (use `dae --version`)**:dae version v0.70
- **OS (e.g `cat /etc/os-release`)**:Debian GNU/Linux 12 (bookworm)
- **Kernel (e.g. `uname -a`)**:Linux rpi 6.6.31+rpt-rpi-v8 #1 SMP PREEMPT Debian 1:6.6.31-1+rpt1 (2024-05-29) aarch64 GNU/Linux
- **Others**:

**Comment by 4860575** [comment]
如果先运行hy2客户端，然后再停止hy2客户端。dae使用“G-CN: 'hysteria2://[1234567890@hy.123.com](mailto:1234567890@hy.123.com):12443'”就可以了。很奇怪的问题

**Comment by 4860575** [comment]
找到问题所在了，将hy2节点的域名改为ip地址，并加上sni=就可以了

**Comment by 0xTimi2233** [comment]
> 找到问题所在了，将hy2节点的域名改为ip地址，并加上sni=就可以了

这个还没修复吗？我现在也遇到这个问题，只能改ip吗？

**Comment by olicesx** [comment]
Closing as outdated: reported against a pre-v2.0 build/environment that has since been superseded by the v2.0.0 / #1099 datapath and control-plane rework, and no longer reproducible against current main. If this still reproduces on the latest release, please open a new issue with current-version logs. Housekeeping after #1099.

**Comment by olicesx** [comment]
Closing as outdated: reported against a pre-v2.0 build/environment that has since been superseded by the v2.0.0 / #1099 datapath and control-plane rework, and no longer reproducible against current main. If this still reproduces on the latest release, please open a new issue with current-version logs. Housekeeping after #1099.
